makefiles/libc/picolibc.mk

```diff
@@ -0,0 +1,23 @@
+ifneq (,$(filter picolibc,$(USEMODULE)))
+  # Test if picolibc.specs is available
+  ifeq ($(shell $(LINK) -specs=picolibc.specs -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
+    USE_PICOLIBC = 1
+    ifeq ($(shell echo "int main(){} void _exit(int n) {(void)n;while(1);}" | LC_ALL=C $(CC) -xc - -o /dev/null -lc -specs=picolibc.specs -Wall -Wextra -pedantic 2>&1 | grep -q "use of wchar_t values across objects may fail" ; echo $$?),0)
+        CFLAGS += -fshort-wchar
+        LINKFLAGS += -Wl,--no-wchar-size-warning
+    endif
+  endif
+endif
+
+ifeq (1,$(USE_PICOLIBC))
+  export LINKFLAGS += -specs=picolibc.specs
+endif
+
+export LINKFLAGS += -lc
+
+ifeq (1,$(USE_PICOLIBC))
+  PICOLIBC_INCLUDE_DIR ?= /usr/local/picolibc/$(TARGET_ARCH)/include
```

Comment:
I think we can do something similar to [`NEWLIB_INCLUDE_DIR`](https://github.com/RIOT-OS/RIOT/blob/6ffac52b1fb6331744466d0a0d134322a1e272d0/makefiles/libc/newlib.mk) here

Reply:
If you use --specs=picolibc.specs, then you don't need to have this definition as the directory will get included automatically. picolibc.specs is installed into the cross-compiler root directory so that you don't need a path for that.

Reply:
Added the `-specs=picolibc.specs` to the `CFLAGS`, thanks for the hint